apps/dashboard/lib/trpc/trpc.ts

```diff
@@ -17,3 +19,38 @@ export const auth = t.middleware(({ next, ctx }) => {
     },
   });
 });
+
+export const protectedProcedure = t.procedure.use(auth);
+
+export const rateLimitedProcedure = ({
+  limit,
+  duration,
+}: {
+  limit: number;
+  duration: number;
+}) =>
+  protectedProcedure.use(async (opts) => {
+    const unkey = new Ratelimit({
```

Comment:
creating this instance every time is a bit ugly

can we create the instances in the file where we specify the limits and then just use a reference here?